### ISSUE TITLE
Count problem

### ISSUE THREAD

**Issue by umatomba** [rescrv/HyperDex#202]
Space configuration with 8 partitions and tolerate 2:

```
space usertable
key k
attributes field0, field1, field2, field3, field4,
           field5, field6, field7, field8, field9
create 8 partitions
tolerate 2 failure
```

Try to insert 1,000,000 records.

Step 1. Count before the test = **0**

```
>>> c = hyperdex.client.Client('192.168.100.1', 1982)
>>> c.count('usertable', {'k': hyperdex.client.Regex('.*')})
0L
```

Step 2. Start sequential insert/read
```
https://github.com/umatomba/hyperdex-test/blob/d86e69ac4ba1860a464f8569d4476609a1d42d03/5nodes/testh/in.c
```

Step 3. Split nodes 4,5 for 30 sec
All inserts was successful. Logs:
```
https://github.com/umatomba/hyperdex-test/blob/master/5nodes/testh/out
```

Step 4. Let's Check It with count

Python client:

```
>>> c.count('usertable', {'k': hyperdex.client.Regex('.*')})
851136L
```

C counter with : 

```
  int64_t hyperdex_client_count(struct hyperdex_client* client,
          const char* space,
          const struct hyperdex_client_attribute_check* checks, size_t checks_sz,
          enum hyperdex_client_returncode* status,
          uint64_t* count);

https://github.com/umatomba/hyperdex-test/blob/master/5nodes/testh/count.c

count = 851136 | op_id: 1 | count_status:HYPERDEX_CLIENT_SUCCESS
```

C checker with sequential read. All records are in DB:

```
https://github.com/umatomba/hyperdex-test/blob/master/5nodes/testh/check.c
check start
check for 1000000 records done
```

**Comment by kaimast** [comment]
Hi umatomba,

I am not sure if this is the problem, but I don't think you are using the C API correctly. There are several problems with your C code.

You don't check the status of your get request. It could have timed out similar (you really should set the timeout value to -1 instead of constantly polling in a while loop). Also when calling loop you should check the opid that is returned. Operations might not be handled in order by hyperdex (as they might be sent to different servers).
Finally, I don't see that you even check if the value was inserted. You do a get request and then never look at the result that is written to attrs/attr_sz.

A small note that isn't really related to hyperdex: don't use magic numbers in your code. Writing something like "op_status != 8448" is really dangerous, because the integer values of the return codes could change anytime. 

Hope this helps.

**Comment by rescrv** [comment]
We've fixed an issue in counts in HyperDex 1.8.1.  Kai's suggestions for improving the code around the C API still stand.
